src/DataAPI.jl

```diff
@@ -287,4 +287,11 @@ using a `sink` function to materialize the table.
 """
 function allcombinations end
 
+"""
+    getmetadata(x, ...)
```

Comment:
How about calling this just `metadata`? The only `get*` methods that exist in Base are those to retrieve a particular element (`getindex`, `getkey`...).

Also, why not allow only a single argument for now? If we want, we can allow passing a second argument giving the name of the metadata to retrieve, but in that case better specify that it should be a string (or a symbol?.)

Reply:
We could call it `metadata`, but the question is if @quinnj would then agree to rename the respective function in Arrow.jl for consistency.

Reply:
> Also, why not allow only a single argument for now?

We could, but I just wanted to signal that `metadata` would dispatch on a first positional argument and that specific types are allowed to opt-in with passing more than one argument if needed. This flexibility is not needed in DataFrames.jl now, but might be needed in the future when we would pass column name or column number as a second argument.

Reply:
> We could call it metadata, but the question is if @quinnj would then agree to rename the respective function in Arrow.jl for consistency.

OK, I hadn't realized that this was the Arrow.jl name. Let's hear what @quinnj thinks.

> We could, but I just wanted to signal that metadata would dispatch on a first positional argument and that specific types are allowed to opt-in with passing more than one argument if needed. This flexibility is not needed in DataFrames.jl now, but might be needed in the future when we would pass column name or column number as a second argument.

Then I would only document the method we expect packages to implement for now, and add more methods later if needed. Having `...` in signatures isn't super useful as it doesn't allow consumers to assume anything about the API.

Reply:
I like just `metadata`